MIDAS/src/systems.cpp

```diff
@@ -208,11 +208,15 @@ bool init_systems(RocketSystems& systems) {
  * Starts thread scheduler to actually start doing jobs
 */
 void begin_systems(RocketSystems* config) {
-    bool success = init_systems(*config);
-    if (!success) {
+    ErrorCode init_error_code = init_systems(*config);
+    if (!(init_error_code == NoError)) {
         // todo some message probably
+        while (true) {
```

Comment:
why put infinte loop in the update function and also here

Reply:
I meant to only put in one place, but forgot I put it in `systems.cpp`, I think leaving the `while (true)` in `systems.cpp` makes the most sense since it is obvious it will hang